js/components/cross-section.tsx

```diff
@@ -1,32 +1,67 @@
 import React from "react";
 import { inject, observer } from "mobx-react";
 import { TransitionGroup, CSSTransition } from "react-transition-group";
-import SmallButton from "./small-button";
-import CrossSection3D from "./cross-section-3d";
+import CrossSection3D, { IOnCreateSceneParams } from "./cross-section-3d";
 import { BaseComponent, IBaseProps } from "./base";
+import { CAMERA_ZOOM_STEP, MAX_CAMERA_ZOOM, MIN_CAMERA_ZOOM } from "../types";
+import ModelCloseIconSVG from "../../images/model-close-icon.svg";
+import ModelResetIconSVG from "../../images/model-reset-icon.svg";
+import ZoomInIconSVG from "../../images/zoom-in-icon.svg";
+import ZoomOutIconSVG from "../../images/zoom-out-icon.svg";
 
 import "../../css/cross-section.less";
 
 export const CROSS_SECTION_TRANSITION_LENGTH = 400; // ms
 
-interface IState {}
+interface IState {
+  cameraZoom: number;
+}
 
 @inject("simulationStore")
 @observer
 export default class CrossSection extends BaseComponent<IBaseProps, IState> {
+  camera: THREE.OrthographicCamera;
+
+  handleCreateScene = (params: IOnCreateSceneParams) => {
+    this.camera = params.camera;
+  };
+
+  handleZoomIn = () => {
+    if (this.camera) {
+      this.camera.zoom = Math.min(MAX_CAMERA_ZOOM, this.camera.zoom + CAMERA_ZOOM_STEP);
+      this.camera.updateProjectionMatrix();
+      this.setState({ cameraZoom: this.camera.zoom });
+    }
+  };
+
+  handleZoomOut = () => {
+    if (this.camera) {
+      this.camera.zoom = Math.max(MIN_CAMERA_ZOOM, this.camera.zoom - CAMERA_ZOOM_STEP);
+      this.camera.updateProjectionMatrix();
+      this.setState({ cameraZoom: this.camera.zoom });
+    }
+  };
+
+  handleResetCamera = () => {
+    this.camera.zoom = 1;
+    this.camera.updateProjectionMatrix();
+    this.simulationStore.resetCrossSectionCamera();
+    this.setState({ cameraZoom: this.camera.zoom });
+  };
+
   render() {
-    const { crossSectionVisible, closeCrossSection } = this.simulationStore;
+    const { crossSectionVisible, showCrossSectionCameraReset, closeCrossSection } = this.simulationStore;
+    const showResetCamera = (this.camera && this.camera.zoom !== 1) || showCrossSectionCameraReset;
```

Comment:
I think that `this.camera.zoom` shouldn't be used in `render()` function, as it isn't React state or MobX observable property. It'll work when `render()` is called for some reason, but it will never itself trigger re-render (and we could expect that looking at this logic).

If you mean using `this.state.zoom`, it'd be more correct, but I think you never update `this.state.zoom` on zoom change initiated by mouse wheel / scrolling.

I think it actually works, as `simulationStore.showCrossSectionCameraReset` becomes true when you zoom in or out. It was supposed to detect only camera angle change, but apparently it also detects zoom by accident (I've checked that in `master` branch).

Reply:
In this implementation, `this.camera.zoom` is the ultimate source of truth, but `this.state.zoom` is updated as a means of triggering re-render. It was admittedly a rather quick-and-dirty solution but putting it in the `simulationStore` as you suggested above is probably the better solution.